priam/src/main/java/com/netflix/priam/connection/CassandraOperations.java

```diff
@@ -202,4 +211,56 @@ public List<Map<String, String>> gossipInfo() throws Exception {
         }
         return returnPublicIpSourceIpMap;
     }
+
+    @Override
+    public List<String> importAll(String srcDir) throws IOException {
+        List<String> failedImports = new ArrayList<>();
+        if (CassandraMonitor.hasCassadraStarted()) {
+            for (Path tableDir : BackupRestoreUtil.getBackupDirectories(srcDir, "")) {
+                String keyspace = tableDir.getParent().getFileName().toString();
+                String table = tableDir.getFileName().toString().split("-")[0];
+                failedImports.addAll(importData(keyspace, table, tableDir.toString()));
+            }
+        } else {
+            recursiveMove(Paths.get(srcDir), Paths.get(configuration.getDataFileLocation()));
+        }
+        return failedImports;
+    }
+
+    private List<String> importData(String keyspace, String table, String source)
+            throws IOException {
+        try (JMXNodeTool nodeTool = JMXNodeTool.instance(configuration)) {
+            return nodeTool.importNewSSTables(
+                    keyspace,
+                    table,
+                    ImmutableSet.of(source),
+                    false /* resetLevel */,
+                    false /* clearRepaired */,
+                    true /* verifySSTables */,
+                    true /* verifyTokens */,
+                    true /* invalidateCaches */,
+                    false /* extendedVerify */,
+                    false /* copyData */);
+        }
+    }
+
+    private void recursiveMove(Path source, Path destination) throws IOException {
+        Preconditions.checkState(Files.exists(source));
+        if (!Files.exists(destination)) {
+            if (!destination.toFile().mkdirs()) {
+                throw new IOException("Failed creating " + destination);
+            }
+        }
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(source)) {
+            for (Path path : directoryStream) {
+                if (Files.isRegularFile(path)) {
+                    Files.move(path, destination.resolve(path.getFileName()));
+                } else if (Files.isDirectory(path)) {
+                    recursiveMove(path, destination.resolve(path.getFileName()));
```

Comment:
If am reading this right, if the source path we are currently iterating over is a directory we recurse by moving the files in that directory into `[destination/directory name]`. Cant we just move the entire directory instead of recursing then?